astropy/utils/decorators.py

```diff
@@ -181,7 +181,7 @@ def deprecate(obj, message=message, name=name, alternative=alternative,
             name = get_function(obj).__name__
 
         altmessage = ''
-        if not message or type(message) == type(deprecate):
+        if not message or isinstance(message, type(deprecate)):
```

Comment:
I'm not sure I understand decorators well enough, but my sense would be to leave this as a `==` -> `is` case; just don't see where subclasses could possibly come in.